File: pe-math/src/main/java/org/saar/maths/transform/ReadonlyRotation.java
package org.saar.maths.transform;

import org.joml.Quaternionfc;
import org.joml.Vector3f;
import org.joml.Vector3fc;
import org.jproperty.ObservableValue;

public interface ReadonlyRotation extends ObservableValue<Quaternionfc> {

    Vector3fc getEulerAngles();

    Vector3f getDirection();

}

File: pe-math/src/main/kotlin/org/saar/maths/transform/ReadonlyRotation.kt
package org.saar.maths.transform

import org.joml.Quaternionfc
import org.joml.Vector3f
import org.joml.Vector3fc
import org.jproperty.ObservableValue

interface ReadonlyRotation : ObservableValue<Quaternionfc> {
    val eulerAngles: Vector3fc
    val direction: Vector3f
}
